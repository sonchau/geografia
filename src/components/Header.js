import React from 'react';

export class Header extends React.Component {

  render () {
    return (
      <header className="header">
      <div className="content-container">
        <div className="header__content">
          <span className="header__message">Welcome </span> 
        </div>
      </div>
    </header>
    )
  }
}

