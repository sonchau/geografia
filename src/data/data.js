export default [
    {
        "sexp_sex": "Male",
        "number": 83156,
        "age5p_age_in_five_year_groups": "0-4 years"
    },
    {
        "sexp_sex": "Female",
        "number": 78572,
        "age5p_age_in_five_year_groups": "0-4 years"
    },
    {
        "sexp_sex": "Male",
        "number": 84278,
        "age5p_age_in_five_year_groups": "5-9 years"
    },
    {
        "sexp_sex": "Female",
        "number": 79873,
        "age5p_age_in_five_year_groups": "5-9 years"
    },
    {
        "sexp_sex": "Female",
        "number": 73504,
        "age5p_age_in_five_year_groups": "10-14 years"
    },
    {
        "sexp_sex": "Male",
        "number": 77299,
        "age5p_age_in_five_year_groups": "10-14 years"
    }     
  ]