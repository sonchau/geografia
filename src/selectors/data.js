
export const buildQuery = (year) => {
    if (year === undefined) {
        return '';
    }
    let query = `SELECT sexp_sex, sum(number) AS number, age5p_age_in_five_year_groups FROM ${year} GROUP BY sexp_sex, age5p_age_in_five_year_groups ORDER BY age5p_age_in_five_year_groups='100 years and over', age5p_age_in_five_year_groups='95-99 years', age5p_age_in_five_year_groups='90-94 years', age5p_age_in_five_year_groups='85-89 years', age5p_age_in_five_year_groups='80-84 years', age5p_age_in_five_year_groups='75-79 years', age5p_age_in_five_year_groups='70-74 years', age5p_age_in_five_year_groups='65-69 years', age5p_age_in_five_year_groups='60-64 years', age5p_age_in_five_year_groups='55-59 years', age5p_age_in_five_year_groups='50-54 years', age5p_age_in_five_year_groups='45-49 years', age5p_age_in_five_year_groups='40-44 years', age5p_age_in_five_year_groups='35-39 years', age5p_age_in_five_year_groups='30-34 years', age5p_age_in_five_year_groups='25-29 years', age5p_age_in_five_year_groups='20-24 years', age5p_age_in_five_year_groups='15-19 years', age5p_age_in_five_year_groups='10-14 years', age5p_age_in_five_year_groups='5-9 years', age5p_age_in_five_year_groups='0-4 years'`;
    return query;
}

export const getLabels = (data) => {
    let groups = data.map((item)=> {
        return item.age5p_age_in_five_year_groups.split(" ")[0]
    })
    return Array.from(new Set(groups))
}

export const getMale = (data) => {
    return data.filter((item)=> {
        return item.sexp_sex === "Male" 
    }).map((item)=> {
        return item.number
    })
}

export const getFemale = (data) => {
    return data.filter((item)=> {
        return item.sexp_sex === "Female" 
    }).map((item)=> {
        return item.number
    })
}

export const getData = (data) => {
    return {
        label: getLabels(data),
        male: getMale(data),
        female: getFemale(data)
    }
}
