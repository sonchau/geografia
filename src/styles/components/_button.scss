.button {
  background: $blue;
  border: none;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
  margin: $s-size;
  cursor: pointer;
}

.button--link {
  background: none;
}

.button--secondary {
  background: #888;
}

button:disabled {
  cursor: default;
  background: #6c757d;
}
