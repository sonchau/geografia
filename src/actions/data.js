import axios from 'axios';
import endPointConfig from './../endpoint/endpoint';
import {buildQuery} from './../selectors/data';

export const setData = (data) => ({
    type: 'SET_DATA',
    data
})

export const startSetData = (year) => {
    const query = buildQuery(year);
    const {endPointURL, endPointURI} = endPointConfig;
    const endpoint = `${endPointURL}${endPointURI}?q=${query}`;

    return(dispatch, getState) => {
        return axios.get(endpoint)
        .then(res => {
            const data = res.data;
            dispatch(setData(data.rows));
        })
        .catch(error => {
            throw(error);
        });
        
    }
}

