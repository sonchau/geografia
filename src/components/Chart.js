import React from 'react';
import ReactDOM from 'react-dom';
import {connect} from 'react-redux';
import {Bar, Line, HorizontalBar} from 'react-chartjs-2';
import {getData} from '../selectors/data';

export class Chart extends React.Component {

    render() {
        const chartData = {
            data: {
                labels: this.props.data.label,
                datasets: [
                {
                    label: 'Male',
                    backgroundColor: 'blue',
                    hoverBackgroundColor: 'rgba(4,12,96,0.4)',
                    data: this.props.data.male
                },
                {
                    label: 'Female',
                    backgroundColor: 'red',
                    hoverBackgroundColor: 'rgba(94,2,2,0.4)',
                    data: this.props.data.female

                }
                ]
            }
        }
        const options = {
            title: {
                display: true,
                text: 'Age Sex Chart',
                fontSize: 40,
            },
            legend: {
                display: true,
                position: 'top'
            }
        }

        return (
            <div className="chart">
            <Bar data={chartData.data} options={options}/>
          </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
      data : getData(state.data)
    }
}

export default connect(mapStateToProps)(Chart);
