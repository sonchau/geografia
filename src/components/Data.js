import React from 'react';
import {connect} from 'react-redux';
import {startSetData} from '../actions/data';
import Chart from './Chart';

export class Data extends React.Component {

    handleOnClick = (props) => {
        this.props.handleOnClick(props.target.value);      
    }

    render(props) {
        return (
            <div className="content-container">
                <Chart />
                <button className="button" value="dlgc_2011_ste_demo_age5gend" onClick={this.handleOnClick}> Year 2011</button>
                <button className="button" value="dlgc_2016_ste_demo_age5gend" onClick={this.handleOnClick}> Year 2016</button>
            </div>    
        )                  
    }
}
const mapDispatchToProps = (dispatch) => ({
    handleOnClick: (year) => dispatch(startSetData(year))
  });
export default connect(undefined, mapDispatchToProps)(Data);