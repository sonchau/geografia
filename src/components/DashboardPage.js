import React from 'react';
import {Header} from '../components/Header';
import Data from '../components/Data';

const DashboardPage = () => (
  <div>
    <Header />
    <Data/>
  </div>
);

export default DashboardPage;
