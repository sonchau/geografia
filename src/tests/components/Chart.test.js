import React from 'react';
import { shallow } from 'enzyme';
import { Chart } from '../../components/Chart';
import {chartData} from '../fixtures/data';

test('should render Chart correctly', () => {
  const wrapper = shallow(<Chart data={chartData} />);
  expect(wrapper).toMatchSnapshot();
});

