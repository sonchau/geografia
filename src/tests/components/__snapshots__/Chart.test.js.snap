// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Chart correctly 1`] = `
<div
  className="chart"
>
  <Bar
    data={
      Object {
        "datasets": Array [
          Object {
            "backgroundColor": "blue",
            "data": Array [
              "1",
            ],
            "hoverBackgroundColor": "rgba(4,12,96,0.4)",
            "label": "Male",
          },
          Object {
            "backgroundColor": "red",
            "data": Array [
              "2",
            ],
            "hoverBackgroundColor": "rgba(94,2,2,0.4)",
            "label": "Female",
          },
        ],
        "labels": undefined,
      }
    }
    options={
      Object {
        "legend": Object {
          "display": true,
          "position": "top",
        },
        "title": Object {
          "display": true,
          "fontSize": 40,
          "text": "Age Sex Chart",
        },
      }
    }
  />
</div>
`;
